src/control/script_command_interface.cpp

```diff
@@ -106,6 +106,114 @@ bool ScriptCommandInterface::setToolVoltage(const ToolVoltage voltage)
   return server_.write(client_fd_, buffer, sizeof(buffer), written);
 }
 
+bool ScriptCommandInterface::startForceMode(const vector6d_t* task_frame, const vector6uint32_t* selection_vector,
+                                            const vector6d_t* wrench, const unsigned int type, const vector6d_t* limits)
+{
+  const int message_length = 26;
+  uint8_t buffer[sizeof(int32_t) * MAX_MESSAGE_LENGTH];
+  uint8_t* b_pos = buffer;
+
+  int32_t val = htobe32(toUnderlying(ScriptCommand::START_FORCE_MODE));
+  b_pos += append(b_pos, val);
+
+  for (auto const& frame : *task_frame)
+  {
+    val = htobe32(static_cast<int32_t>(frame * MULT_JOINTSTATE));
+    b_pos += append(b_pos, val);
+  }
+
+  for (auto const& selection : *selection_vector)
+  {
+    val = htobe32(static_cast<int32_t>(selection * MULT_JOINTSTATE));
+    b_pos += append(b_pos, val);
+  }
+
+  for (auto const& force_torque : *wrench)
```

Comment:
@shuobh I'm afraid, I don't quite understand your comment here.

Reply:
UR people once told me that zero_ft_sensor will be called when new script command is received. We have been working with some applications where I need to zero_ft_sensor when no external force exists and use multiple force_mode to control the robot, i.e., z+ with 5N for 5s, yaw+ with 1N for 5s. If the command is sent within the same script function, for example, force_mode(z+ with 5N) -> sleep 5s -> force_mode(yaw+ with 1N) -> sleep 5s, then external force compare to the beginning is accounted for. With the command being sent one by one, only the external force before and after each command is accounted for. For example, if after force_mode(z+ with 5N) -> sleep 5s, the robot is in contact with and an external force of z- with 1N is applied, then this will effect force_mode(yaw+ with 1N).

Reply:
Thanks for following up.

I don't think this is applicable here, since we are not running different scripts. This is all running inside the same script, where `force_mode` commands are injected while the script is running.